### ISSUE TITLE
Shibboleth: Institution Log In changes: Help link and extra text

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#2951]
The Help button below goes to https://demo.dataverse.org/guides/user/account.html and instead it should go to http://guides.dataverse.org/en/latest/user/account.html

![log_in_-_demo_dataverse_-_2016-02-12_14 05 42](https://cloud.githubusercontent.com/assets/21006/13017398/f9d05b76-d191-11e5-8630-2f7e6efd037a.png)

Here's the code:

```
murphy:dataverse pdurbin$ ack '/guides/user/account.html' src
src/main/webapp/resources/js/shib/idpselect_config.js
19:    this.helpURL = '/guides/user/account.html';
```

**Comment by pdurbin** [comment]
Incidentally, on the server hosting https://demo.dataverse.org I just moved the old locally-deployed guides aside so that errors like this are no longer masked: `cd /usr/local/glassfish4/glassfish/domains/domain1/docroot/ && mv guides guides.moved-because-all-links-should-go-to-guides.dataverse.org`

We should probably do the same for production because an old stale version from the 4.0 release is still at https://dataverse.harvard.edu/guides/index.html

**Comment by pdurbin** [comment]
@eaquigley I fixed the link in 4121376 so at least it will go **somewhere** but as of this writing (4.2.4) http://guides.dataverse.org/en/latest/user/account.html doesn't say anything about "institutional login" and actually has bad advice for changing your password and editing your name, etc... if you're a Shib user. Since this is a User Guide issue at this point I'm going to pass this issue to you for consideration.

I'm tracking this issue in the parent issue at #2939 so it doesn't fall through the cracks.

**Comment by bencomp** [comment]
Shouldn't the link be the same as the (configurable) Support link at the top of the page, or a deeplink to the Institutional login information at the configurable location?

**Comment by pdurbin** [comment]
@bencomp good point that there's already a "Support" link at the top of every page. We could just hide the "Help" link all together. I just don't want people to see this (broken link to non-existent locally-deployed guides):

![screen shot 2016-02-23 at 9 50 37 am](https://cloud.githubusercontent.com/assets/21006/13255209/01126a9e-da13-11e5-887c-6226aa2ebb6c.png)

@eaquigley do you have any opinions on this? I put this in your name because I thought you might want to add some stuff about Shibboleth to the User Guide but that probably should be tracked in a different issue. In this one I'm simply trying to fix a broken link. Please feel free to pass this back to me if you think the fix so far at https://github.com/IQSS/dataverse/issues/2951#issuecomment-183464501 is good enough. It fixes the link at least even if there's no content in the User Guide about Shibboleth.

**Comment by eaquigley** [comment]
@pdurbin I plan on adding Shibb info into the user guide however don't know which release it will be in so I don't want to start a documentation branch for it until then. 

Regarding the Help link, I like @bencomp's idea. How feasible is that @pdurbin?

**Comment by pdurbin** [comment]
@bencomp actually seems to have two different ideas but the commonality is that rather than being a hard-coded link he wants it to be dynamic.

I see a few downsides with the first idea (support link popup):

- It's further hacking on the fork we made of the [Embedded Discovery Service (EDS)](https://shibboleth.net/products/embedded-discovery-service.html) JavaScript that @mheppler helped me hack on in #794 . Ideally we'd abandon our fork someday and switch to the official build.
- EDS supports internationalization and we'd lose that.
- I don't know how to call into SendFeedbackDialog backing bean from Javascript ( @scolapasta or @sekmiller or @mheppler or @raprasad any ideas?). I spoke with @landreev and I'm guessing it's something like `PF('contactForm').show()`.

The second idea is to have a configurable URL. We would still need a sane default, however, which probably puts us back at http://guides.dataverse.org/en/latest/user/account.html (or a specific shib page) as a sane default. Docs need to be written, etc. To let people change the default URL I **think** we could use a trick similar to `#{settingsWrapper.get(':GoogleAnalyticsCode')}` but I'm not sure.

@eaquigley bottom line is both ideas seem like maybe more work than it's worth. I'm not sure which one you meant. Also, for Shib docs, I think you should use https://github.com/IQSS/dataverse/tree/2939-shib since that's where @mheppler and I are working on Shib-related work. Please note that for now Shibboleth is marked as "experimental" in the Installation Guide so I assume it should be marked as "experimental" in the User Guide as well (for now). My plan is to run Shibboleth in production at Harvard for a while and **then** recommend we remove "experimental" from the guides.

As for timing of phase 1 Shibboleth stuff (#2939), I have until Friday to make a pull request for 4.3. Otherwise, it will be a 4.4 thing.

Does this help? Please advise. Also, can you please make an issue for updating the User Guide to include Shibboleth?

**Comment by eaquigley** [comment]
@pdurbin please take the Help link out and right below Institution Log In header add this: 

Use Dataverse with your institutional log in instead of creating an account. Learn more {link to guides}.

I haven't created the guides section for Shibb yet though so that link can't be populate :( Working on it now.

**Comment by pdurbin** [comment]
@eaquigley I made those two changes (minus a link to the Guides until you're ready). I deployed the latest to https://shibtest.dataverse.org so you can check it out and I'll give you this issue back until you're ready for me to add a link to the guides. You're welcome to pass this to @mheppler if you'd like him to rearrange the new text I added to the Log In page.

**Comment by eaquigley** [comment]
Thanks @pdurbin! I'll pass to @mheppler to spruce that text up once I have the shib guide link.

**Comment by mheppler** [comment]
Revised text for Shibboleth users on the Account Information pg, added a link to the Contact Dataverse Support popup. Also added this link to the error message text for 404 and 403 errors.

**Comment by pdurbin** [comment]
@eaquigley this issue is technically blocked on #2975 (which is already in your name) if you still want a link to a section on remote auth in the User Guide. I've already vetted @mheppler various changes so I'm going to mark this issue as code complete and put it in my name to write up a summary for QA. If you still want the link, please work on #2975 and let me know when you're ready for the link to be added. Otherwise, we'll ship without the link.

**Comment by eaquigley** [comment]
@pdurbin I'm almost done with the user guide. it will be ready early next week. if this isn't being QA'd till later next week (which i'm assuming is the case since @kcondon will be out and also needs to finish testing 4.3), then this blocker will go away before QA.

**Comment by pdurbin** [comment]
@eaquigley fantastic! I'll keep an eye on #2975 for updates!

**Comment by pdurbin** [comment]
I believe that #2975 is far enough along that I'm no longer blocked on this issue so I'll add the link to User Guide now that there's a section on Shib in pull request #3025.

**Comment by pdurbin** [comment]
> Use Dataverse with your institutional log in instead of creating an account. Learn more {link to guides}.

@eaquigley in 0a5a677 I added in the "Learn More" link. I just deployed this code to https://shibtest.dataverse.org if you want to check it out. Here's a screenshot:

<img width="580" alt="screen shot 2016-03-21 at 2 22 55 pm" src="https://cloud.githubusercontent.com/assets/21006/13929523/e8a98bf4-ef70-11e5-9eb6-1790b898695c.png">

@mheppler I spent more time on this than I thought I would have to. In the end I split up the Bundle key into two keys (blurb vs. support) because otherwise the support popup stopped working. With two keys it seems to work fine. If you want to hack on this any more, please let me know because I'm planning on writing up a summary for QA right now basically. Thanks.

**Comment by pdurbin** [comment]
Passing to QA. To summarize the changes to the login page made in pull request #3025:

- Link labeled "Help" removed.
- New text with two links.
  - "Learn More" goes to "account" page in the User Guide
  - Support link opens a popup form.

**Comment by kcondon** [comment]
Support popup bot check math function is missing operands. 2+3= appears as +=. Help link is removed.

**Comment by pdurbin** [comment]
@kcondon unfortunately, the math challenge started being broken in 4.3 and it's gotten worse in the Shib branch. I just opened #3036 about this. Thanks for the heads up.

**Comment by pdurbin** [comment]
@scolapasta I'm giving this issue to you. Please see my comment at https://github.com/IQSS/dataverse/issues/3036#issuecomment-202439945 .

**Comment by kcondon** [comment]
OK, since the only remaining issue in this ticket was opened in a new ticket, 3036 (thanks phil) we can close this one. The 4.3 issue is only on sign up welcome screen, was not aware of changes there. Noticed it here because shib changes added more places for this link and so triggered explicit test.

Closing.
